test/src/streams/streamsTest.cpp

```diff
@@ -0,0 +1,68 @@
+#include <RGLFields.hpp>
+#include <gtest/gtest.h>
+#include <utils.hpp>
+#include "testKernel.hpp"
+
+using namespace testing;
+
+class StreamsTest : public RGLAutoCleanupTest {
+protected:
+};
+
+TEST_F(StreamsTest, basic_streams)
+{
+    // Prepare two non-default streams.
+    cudaStream_t graphStream;
+    cudaStream_t copyStream;
+
+    cudaStreamCreate(&graphStream);
+    cudaStreamCreate(&copyStream);
+
+    int N = 20000;
+    size_t size = N * sizeof(float);
+
+    // Allocate input vector h_A in host memory
+    float* h_A = (float*)malloc(size);
+
+    // Initialize input vectors
+    for (int i = 0; i < N; ++i) {
+        h_A[i] = i;
+    }
+    // Create event for sensitive job (kernel) end.
+    cudaEvent_t jobDoneEvent;
+    // Create event without timer (used for profiling).
+    // This way, we have got some speed, and also according to CUDA documentation, this flag is neccessary to make sure that multi-stream synchronization is bulltetproof.
+    cudaEventCreateWithFlags(&jobDoneEvent, cudaEventDisableTiming);
+
+    // Copy vector from host memory to device memory
+    float* d_A;
+    cudaMallocAsync(&d_A, size, graphStream);
+    cudaMemcpyAsync(d_A, h_A, size, cudaMemcpyHostToDevice, graphStream);
+
+    // Run sensitive job that we forecast, it takes some time.
+    testKernelWrapper(N, d_A, graphStream);
+
+    // Record synchronization event to the stream after sensitive job.
+    cudaEventRecord(jobDoneEvent, graphStream);
+
+    // Synchronize streams. This is pass or not for the test.
+    //  All API Calls in given stream, will wait for the event  to be completed.
+    cudaStreamWaitEvent(copyStream, jobDoneEvent);
```

Comment:
Here, I'd also consider adding some more work to the graph stream after the event (e.g. dummy `cudaMallocAsync`, `cudaMemsetAsync`, `cudaFreeAsync`. It's hard to do any additional checks, but it makes graph stream load more realistic (simulating other nodes). Also, it will allow us to make sure (using NVIDIA Nsight) that the work in the graph streams continues when copy stream works.